src/ring_theory/discrete_valuation_ring.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2020 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard
+-/
+
+import ring_theory.principal_ideal_domain order.conditionally_complete_lattice
+import ring_theory.multiplicity
+import ring_theory.valuation.basic
+import tactic
+
+/-!
+# Discrete valuation rings
+
+This file defines discrete valuation rings (DVRs) and develops a basic interface
+for them.
+
+## Important definitions
+
+There are various definitions of a DVR in the literature; we define a DVR to be a local PID
+which is a not a field (the first definition in Wikipedia) and prove that this is equivalent
+to being a PID with a unique non-zero prime ideal (the definition in Serre's
+book "Local Fields").
+
+Let R be an integral domain, assumed to be a principal ideal ring and a local ring.
+
+* `DVR R` : a predicate expresing that R is a DVR
+
+### Notation
+
+It's a theorem that an element of a DVR is a uniformiser if and only if it's irreducible.
+We do not hence define `uniformiser` at all, because we can use `irreducible` instead.
+
+### Definitions
+
+## Implementation notes
+
+## Tags
+
+discrete valuation ring
+-/
+
+open_locale classical
+
+universe u
+
+open ideal local_ring
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+/-- A commutative ring is a discrete valuation ring if it's a local PID which is not a field -/
```

Comment:
```suggestion
/-- An integral domain is a discrete valuation ring if it's a local PID which is not a field -/
```
To match the fact that `integral_domain` is a typeclass parameter?

Reply:
Yeah, I like that. Of course I'd bundled the ID assertion into the PID assertion, but it's perhaps clearer your way.